lib/features/loan/source/api_source.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'package:logger/logger.dart';
+
+final Logger logger = Logger();
+Future<int> getRandomNumber() async {
+  final url = Uri.parse(
```

Comment:
Do this with RestApi. Here is one random class from my project that uses RestApi
```dart
import 'dart:convert';

import 'package:dio/dio.dart';
import 'package:injectable/injectable.dart';
import 'package:pretty_dio_logger/pretty_dio_logger.dart';
import 'package:retrofit/retrofit.dart';

part 'mux_data_source.g.dart';

@RestApi(parser: Parser.MapSerializable)
@lazySingleton
abstract class MuxDataSource {
  @factoryMethod
  factory MuxDataSource(FlavorConfig flavorConfig) {
    return _MuxDataSource(Dio(), baseUrl: 'https://api.mux.com/video/v1');
  }

  @GET('/uploads/{uploadId}')
  Future<MuxResponse<GetUploadResponseData>> getUpload(@Path() String uploadId);
}

```
I think I already wrote this one somewhere